packages/core/docs/integrate/cms.md

```diff
@@ -51,36 +51,47 @@ The factory will output a composable fulfilling our interfaces. That's all you h
 
 ### Content rendering (optional)
 
-Many CMS systems allow to control the page layout by returning a list of components as a JSON file. These components are then rendered in the applicatio. The component that is mapping the `content` object into Vue components and renders them is called `RenderContent`.
+Many CMS systems allow to control the page layout by returning a list of components as a JSON file. These components are then rendered in the application. The component that is mapping the `content` object into Vue components and renders them is called `RenderContent`.
 
 ```html
-<!-- public API -->
 <RenderContent :content="content">
 ```
 
-To build this component you just have to use our `renderComponentFactory`.
+Now you need to prepare special structure for your components. You can create function, helper to filter all the metadata that you've fetched from the CMS and just leave the name and props.
 
-```ts
-// integration
-import { renderComponentFactory } from '@vue-storefront/core'
-
-// CONTENT and CONTENT_SEARCH_PARAMS are your CMS-specific types, we advise to have at least 'id' param for search
-const RenderComponent = renderComponentFactory({
-  // (extractContent: CONTENT) => { componentName, props }[]
-  extractContent (content) { 
-    // extract an array of { componentName: string, props: { name, value } } pairs and the factory will generate a component that will render these components based on the ones that are currently registered within application
-    return components[]
-  }
-}) 
+```typescript
+components: {
+  componentName: string
+  props: {}
+}[]
 ```
-Under the hood the `RenderComponent` is rendering components using Vue dynamic component feature:
 
-```html
-<!-- core, inside renderComponentFactory -->
-<component v-for="component in components" is="component.componentName" content="props" />
+Pass it to the `RenderContent` component as a `content` prop. To register components use computed value.
+
+```vue
+<!-- inside the RenderContent.vue component -->
+<component v-for="component in components" :key="index" is="component.componentName" v-bind="component.props" />
+
+<script>
+export default ({
+  name: 'RenderContent',
+  props: {
+    content: {
+      type: Array,
+    },
+  },
+  computed: {
+    components() {
+      return yourDataBuildFunction(this.content)
```

Comment:
imho you should suggest a name here (`extractComponents`), also you're missing the import